src/csv_upload/csvUpload.js

```diff
@@ -0,0 +1,217 @@
+            testComment=new Comment(commentType);//create an address instance
+            }
+            else if(headerElements[0]=="charges"){
+            testComment=new Charge(commentType);//create an address instance   
+            }
+            else if (headerElements[0]=="codes"){
+            testComment=new Code(commentType);//create an address instance
+           }
+       
+            var commentObject={};
+            commentObject[headerElements[1]]=testComment;
+            finalObject[headerElements[0]]=commentObject;
+          }
+          else if(references.includes(headerElements[0]) && !references.includes(headerElements[1]))
+          {
+             var commentType={}; 
+             commentType[headerElements[2]]=currentValue;//get the string before the first ','
+
+            let testComment;
+            if(headerElements[0]=="addresses"){
+            testComment=new Address(commentType);//create an address instance
+            }
+            else if(headerElements[0]=="emails"){
+            testComment=new Email(commentType);//create an address instance
+            }
+            else if(headerElements[0]=="references"){
+            testComment=new Reference(commentType);//create an address instance
+            }
+            else if(headerElements[0]=="comments"){
+             testComment=new Comment(commentType);//create an address instance
+            }
+            else if(headerElements[0]=="charges"){
+             testComment=new Charge(commentType);//create an address instance   
+            }
+            else if(headerElements[0]=="codes"){
+            testComment=new Code(commentType);//create an address instance
+            }
+             finalObject[headerElements[0]][headerElements[1]]=testComment;
+          }
+          else if (references.includes(headerElements[0]) && references.includes(headerElements[1]))
+          {
+             finalObject[headerElements[0]][headerElements[1]][headerElements[2]]=currentValue;   
+          }
+          references=headerElements;//update references
+        }
+      else
+      {
+        var convertedHeader=camelCase(headers[i]); //convert headers to camelcase      
+        if(currentLine==null) //the last value is not specified
+        {
+          finalObject[convertedHeader]=undefined;
+          continue;
+        }
+
+        var firstPart=currentLine.split(/,(.+)/)[0];//get the string before the first ','
+        if(firstPart.charAt(firstPart.length-1)==",") firstPart=firstPart.substr(0, firstPart.length-1);
+        currentLine=currentLine.split(/,(.+)/)[1];
+
+        if(firstPart=="") 
+        {
+           finalObject[convertedHeader]=undefined;
+           continue;
+        }
+        
+        if(firstPart=="true") firstPart=true;
+        else if(firstPart=="false") firstPart=false;    
+        else if(!isNaN(firstPart))
+        firstPart= parseInt(firstPart);
+        finalObject[convertedHeader]=firstPart;
+      }
+    }
```

Comment:
The `finalObject` will then have to be converted into a Transaction (or Asset or Party or Book). Based on the `type` parameter supplied. E.g. if type == 'transactions' then `finalObject = new Transaction(finalObject)`.